src/dotnet/Chat.UI.Blazor/Components/InviteChatAuthor.razor

```diff
@@ -0,0 +1,139 @@
+    [Inject] private IUserAuthors UserAuthors { get; init; } = null!;
+    [Inject] private IChatAuthors ChatAuthors { get; init; } = null!;
+
+    [Parameter] public Chat Chat { get; set; } = null!;
+    [CascadingParameter] BlazoredModalInstance ModalInstance { get; set; } = null!;
+
+    private ContactViewModel[] FilteredChatContacts { get; set; } = Array.Empty<ContactViewModel>();
+    private ContactViewModel[] AllChatContacts { get; set; } = Array.Empty<ContactViewModel>();
+    private bool CanInvite { get; set; }
+
+    protected override async Task OnInitializedAsync() {
+        var existingChatUserIds = new HashSet<string>(
+            await ChatAuthors.GetUserIds(Session, Chat.Id, CancellationToken.None),
+            StringComparer.Ordinal);
+
+        var contacts = await UserContacts.GetContacts(Session, CancellationToken.None);
+        var resolveUserAuthors = contacts.Select(x => UserAuthors.Get(x.TargetUserId, true, CancellationToken.None));
+        var userAuthors = await Task.WhenAll(resolveUserAuthors);
+        var chatContacts = userAuthors
+            .Where(x => x != null)
+            .Select(x => x!)
+            .Select(contact => {
+                if (!string.IsNullOrWhiteSpace(contact.Picture)) {
+                    return contact;
+                }
+
+                var picture = $"https://avatars.dicebear.com/api/avataaars/{contact.Name}.svg";
+                return contact with {
+                    Picture = picture,
+                };
+            })
+            .Select(x => new ContactViewModel {
+                Existing = existingChatUserIds.Contains(x.Id),
+                Checked = existingChatUserIds.Contains(x.Id),
+                Author = x,
+            })
+            .ToArray();
+        AllChatContacts = chatContacts;
+        FilteredChatContacts = chatContacts;
+    }
+
+    private async Task OnInvite() {
+        CanInvite = false;
+
+        var userIds = AllChatContacts
+            .Where(x => !x.Existing && x.Checked)
+            .Select(x => x.Author.Id.Value)
+            .ToArray();
+        var createChatAuthorsCommand = new IChatAuthors.CreateChatAuthorsCommand(Session, Chat.Id, userIds);
+        await ChatAuthors.CreateChatAuthors(createChatAuthorsCommand, CancellationToken.None);
+
+        await ModalInstance.CloseAsync();
+    }
+
+    private async Task OnCancel() {
+        await ModalInstance.CancelAsync();
+    }
+
+    private void OnFilter(ChangeEventArgs args) {
+        var filter = args.Value?.ToString();
+        if (string.IsNullOrWhiteSpace(filter)) {
+            FilteredChatContacts = AllChatContacts.ToArray();
+        } else {
+            FilteredChatContacts = AllChatContacts
+                .Where(x => x.Author.Name.Contains(filter, StringComparison.OrdinalIgnoreCase))
+                .ToArray();
+        }
+    }
+
+    private void OnContactClick(ContactViewModel contact) {
+        if (contact.Existing) {
```

Comment:
We don't use brackets for a single-statement block -- unless you need to put a comment inside as well :)

Reply:
done